### PR TITLE
Updates to making QC Summary files

### PR DESCRIPTION
**Pull request by kenkehoe**
These are some incremental updates to the method that creates a QC Summary. Also, working on improving the output netCDF file by improving how we correct the Dataset when writing to follow CF. Added some pytest.warn() to capture warnings and ensure they occur in our testing.

- [X ] Tests added
- [ X] Documentation reflects changes
- [ X] PEP8 Standards or use of linter
- [X ] Xarray Dataset or DataArray variable naming follows 'ds' or 'da' naming

### CODE REVIEW

**Comment by zssherman** [comment]
@kenkehoe Thanks for the PR, I'll give this a review tomorrow

File: act/qc/qc_summary.py [hunk 1]
```diff
@@ -91,34 +105,54 @@ def create_qc_summary(self, cleanup_qc=False):
...
                 qc_mask = self.get_masked_data(
                     var_name, rm_assessments=assessment, return_mask_only=True
                 )
 
-                # Do not really know how to handle scalars yet.
-                if qc_mask.ndim == 0:
-                    continue
+                # # Do not really know how to handle scalars yet.
+                # if return_ds[var_name].ndim == 0:
+                #     warnings.warn(f'Unable to process scalar variable {var_name}. '
```
Comment (zssherman) [inline act/qc/qc_summary.py:125]: Can we remove this commented out code as well?

**Review by zssherman — CHANGES_REQUESTED** [review]
Made some minor code requested changes, but besides that looks good to me

File: act/qc/arm.py [hunk 2]
```diff
@@ -68,6 +70,9 @@ def add_dqr_to_qc(
     skip_location_vars : boolean
         Does not apply DQRs to location variables.  This can be useful in the event
         the submitter has erroneously selected all variables.
+    create_missing_qc_variables : boolean
+        If a quality control varible for the data varialbe does not exist,
```
Comment (zssherman) [inline act/qc/arm.py:74]:
```suggestion
        If a quality control variable for the data variable does not exist,
```

File: act/qc/arm.py [hunk 3]
```diff
@@ -102,8 +107,35 @@ def add_dqr_to_qc(
     if cleanup_qc:
         ds.clean.cleanup()
 
-    start_date = ds['time'].values[0].astype('datetime64[s]').astype(dt.datetime).strftime('%Y%m%d')
-    end_date = ds['time'].values[-1].astype('datetime64[s]').astype(dt.datetime).strftime('%Y%m%d')
+    # Get time from Dataset
+    time = ds['time'].values
+
+    # If the time is not a datetime64 because the read routine was not asked to
+    # convert CF variables, convert the time varible for this routine only.
```
Comment (zssherman) [inline act/qc/arm.py:114]:
```suggestion
    # convert CF variables, convert the time variable for this routine only.
```

File: act/qc/arm.py [hunk 4]
```diff
@@ -181,6 +213,10 @@ def add_dqr_to_qc(
             if skip_location_vars and var_name in loc_vars:
                 continue
 
+            # Do not process time varibles
```
Comment (zssherman) [inline act/qc/arm.py:216]:
```suggestion
            # Do not process time variables
```

File: act/qc/clean.py [hunk 5]
```diff
@@ -792,7 +792,8 @@ def normalize_assessment(
         self,
         variables=None,
         exclude_variables=None,
-        qc_lookup={'Incorrect': 'Bad', 'Suspect': 'Indeterminate'},
+        # qc_lookup={'Incorrect': 'Bad', 'Suspect': 'Indeterminate'},
```
Comment (zssherman) [inline act/qc/clean.py:795]:
```suggestion
```

File: act/qc/qc_summary.py [hunk 6]
```diff
@@ -91,34 +105,54 @@ def create_qc_summary(self, cleanup_qc=False):
...
 
                 qc_mask = self.get_masked_data(
                     var_name, rm_assessments=assessment, return_mask_only=True
                 )
 
-                # Do not really know how to handle scalars yet.
-                if qc_mask.ndim == 0:
-                    continue
+                # # Do not really know how to handle scalars yet.
+                # if return_ds[var_name].ndim == 0:
```
Comment (zssherman) [inline act/qc/qc_summary.py:124]:
```suggestion
```

File: act/qc/qc_summary.py (hunk 1 above)
Comment (zssherman) [inline act/qc/qc_summary.py:125]:
```suggestion
```

File: act/qc/qc_summary.py [hunk 7]
```diff
@@ -91,34 +105,54 @@ def create_qc_summary(self, cleanup_qc=False):
...
+                #     warnings.warn(f'Unable to process scalar variable {var_name}. '
+                #                   'Scalar variables currently not implemented.')
```
Comment (zssherman) [inline act/qc/qc_summary.py:126]:
```suggestion
```

File: act/qc/qc_summary.py [hunk 8]
```diff
@@ -91,34 +105,54 @@ def create_qc_summary(self, cleanup_qc=False):
...
+                #     continue
```
Comment (zssherman) [inline act/qc/qc_summary.py:127]:
```suggestion
```

**Review by zssherman — CHANGES_REQUESTED** [review]

**Comment by zssherman** [comment]
@kenkehoe I made some changes etc, are you good with me merging this in?

**Comment by kenkehoe** [comment]
Yes, you can merge your changes.

**Review by AdamTheisen — APPROVED** [review]

**Comment by AdamTheisen** [comment]
@zssherman this looks good to me.  If all changes you requested have been implemented, I would say merge away!

**Review by mgrover1 — APPROVED** [review]

**Pull request merged**